### DISCUSSION TITLE
Fix alignment for Fugitive blame

### DISCUSSION THREAD

**Discussion by ouuan** [Bekaboo/dropbar.nvim#153] [Show and tell]
By default, Fugitive blame does not have the drop bar, so the line blames are not aligned with the file content.

![blame not aligned](https://github.com/Bekaboo/dropbar.nvim/assets/30581822/f2921c99-552d-4f98-bb3b-891df4508e7b)

I use the following config to add a dummy drop bar to fix it:

```lua
local default_config = require 'dropbar.configs'.opts

require 'dropbar'.setup {
  general = {
    enable = function(buf, win)
      return default_config.general.enable(buf, win)
        or vim.bo[buf].ft == 'fugitiveblame'
    end,
  },
  bar = {
    sources = function(buf, win)
      if vim.bo[buf].ft == 'fugitiveblame' then
        return {}
      end
      return default_config.bar.sources(buf, win)
    end
  },
}
```

![fixed](https://github.com/Bekaboo/dropbar.nvim/assets/30581822/5a4fa08a-6339-43d1-8429-4fdb940f5063)

**Comment by feakuru** [comment]
stumbled upon this while looking to solve exactly this problem, thank you ouuan!
the configs have been seemingly changed a bit recently, here's the version that worked for me:
```lua
local default_config = require 'dropbar.configs'.opts
require('dropbar').setup {
  bar = {
    enable = function(buf, win)
      return default_config.bar.enable(buf, win)
        or vim.bo[buf].ft == 'fugitiveblame'
    end,
    sources = function(buf, win)
      if vim.bo[buf].ft == 'fugitiveblame' then
        return {}
      end
      return default_config.bar.sources(buf, win)
    end
  },
}
```

**Comment by Bekaboo** [comment]
@feakuru A more general method that doesn't depend on dropbar's setup function is:

```lua
vim.api.nvim_create_autocmd('FileType', {
  desc = 'Set buffer-local options for fugitive blame buffers.',
  group = vim.api.nvim_create_augroup('FugitiveSettings', {}),
  pattern = 'fugitiveblame',
  callback = function()
    local win_alt = vim.fn.win_getid(vim.fn.winnr('#'))
    vim.opt_local.winbar = vim.api.nvim_win_is_valid(win_alt)
        and vim.wo[win_alt].winbar ~= ''
        and ' '
      or ''
  end,
})
```

Put this in your fugitive config and it will work even if you switch to other winbar plugins or when the source window does not have winbar enabled.
